### ISSUE TITLE
[Bug]: torchdynamo.export doesn't work with module.named_buffers

### ISSUE THREAD

**Issue by tugsbayasgalan** [pytorch/torchdynamo#1738]
### 🐛 Describe the bug

Following doesn't work: 

```
class D(torch.nn.Module):
    def __init__(self):
        super().__init__()

    def forward(self, a):
        count = 0
        for i in self.named_buffers():
            count += 1
        return a.cos()

torchdynamo.config.dynamic_shapes = True
torchdynamo.export(D(), torch.randn(3, 4), aten_graph=True)
```

### Error logs
~~~
  File "/mnt/xarfuse/uid-23137/7756fc57-seed-nspid4026533012_cgpid17879168-ns-4026533009/torchdynamo/variables/nn_module.py", line 401, in call_method
    return super().call_method(tx, name, args, kwargs)
  File "/mnt/xarfuse/uid-23137/7756fc57-seed-nspid4026533012_cgpid17879168-ns-4026533009/torchdynamo/variables/base.py", line 249, in call_method
    raise unimplemented(f"call_method {self} {name} {args} {kwargs}")
  File "/mnt/xarfuse/uid-23137/7756fc57-seed-nspid4026533012_cgpid17879168-ns-4026533009/torchdynamo/exc.py", line 71, in unimplemented
    raise Unsupported(msg)
torchdynamo.exc.Unsupported: call_method NNModuleVariable() named_buffers [] {}
~~~
### Did Dynamo succeed?

- [ ] Does dynamo.optimize("eager") succeed?

### Did AOT succeed?

- [ ] Did dynamo.optimize("aot_eager") succeed?

### Did Inductor succeed?

- [ ] Does dynamo.optimize("inductor") succeed?

### Minified repro

Repro above
